allennlp/nn/initializers.py

```diff
@@ -39,20 +72,27 @@ class InitializerApplicator:
     All parameters in the Module will be initialized.
     """
     def __init__(self,
-                 initializers: Dict[str, Callable[[torch.Tensor], None]] = None,
-                 default_initializer: Callable[[torch.Tensor], None] = torch.nn.init.normal) -> None:
+                 initializers: Dict[str, Initializer] = None,
+                 default_initializer: Initializer = Initializer.by_name('normal')(),
+                 exclude: Sequence[str] = None) -> None:
         """
         Parameters
         ----------
-        initializers : Dict[str, Callable[[torch.Tensor], None]], optional (default = {})
+        initializers : ``Dict[str, Callable[[torch.Tensor], None]]``, optional (default = {})
             A dictionary mapping parameter regexes to initializers to be applied to parameters
             matching the regex.
-        default_initializer : Callable[[torch.Tensor], None], optional, (default = torch.nn.init.normal)
+        default_initializer : ``Callable[[torch.Tensor], None]``, optional (default = torch.nn.init.normal)
             A default initializer, which will be used in the case that the Applicator encounters a parameter
             which does not match any of the regexes provided.
+        exclude : ``Sequence[str]``, optional (default=``[]``)
```

Comment:
The question for me is what's easier to understand and use?  If you have competing regexes around which things get initialized with what, it just seemed too confusing.  It's a lot simpler if you just say that exclusion only applies to things that would have been initialized with the default.